Add grass crate to project and implement SCSS to css conversion

humbamp123 opened the pull request.

humbamp123 committed:
Add grass crate to project and implement SCSS to css conversion

caer commented on test/simple_example.scss:
Can you move this file to `tests/test.scss`? I generally house test assets within the top-level `tests` directory.

caer commented on src/asset/scss.rs:
Can you run `cargo fmt` (code formatting) and `cargo clippy` (code listing) on this branch? This is my bad--I heavily procrastinated on setting up a CONTRIBUTING.md, and _just_ now pushed one up for another repo: https://github.com/with-caer/OwC-workbench/blob/main/CONTRIBUTING.md 🙈

caer commented on src/asset/scss.rs:
```diff
@@ -0,0 +1,42 @@
+        let css = from_path(Path::new(path), &Options::default()).unwrap();
+
+        // Update the asset's contents and target extension.
+        let text = asset.contents.try_as_mut_text().expect("todo");
```
This is more of a note than a change request, but: we'll need a fast-follow PR to update the contract for `ProcessesAssets` to return a `Result<(), Error>`: We don't want to be using `unwrap()` or `expect()`, and should instead be returning an error (`unwrap` and `expect` will `panic!`, crashing the app).

humbamp123 replied:
I was going to NOT use expect/unwrap, but I followed your lead on the MarkdownProcessor....

caer replied:
Indeed, I've set a poor example... 🙈

caer commented on src/asset/scss.rs:
```diff
@@ -0,0 +1,42 @@
+
+    #[test]
+    fn processes_scss() {
+        let mut scss_asset = Asset::new("src/test/test.scss".into(), "".as_bytes().to_vec());
```
Would you be able to expand the test case such that this SCSS file is referencing one other file on-disk, just to make sure we're properly configuring/using grass' internal importing logic?

caer reviewed (changes requested):
This is a great PR; thank you so much for working on it!! I've requested a few changes, but they're mostly little things that are my bad for not having a CONTRIBUTING.md in place yet. 🙈

humbamp123 committed:
Resolve PR comments
- Added a test with nested scss
- Moved test folder to main directory
- Ran formatter/clippy/tests
